app/server.go

```diff
@@ -334,6 +334,9 @@ func NewServer(options ...Option) (*Server, error) {
 		}
 	}
 
+	s.SearchEngine.UpdateConfig(s.Config())
```

Comment:
Because it depends on the implementation of each backend, maybe you want to run or not the initialization process when you create the broker and register the backends, but not when you change the config, each backend can behave different depending on if you are initializing or updating the config, some config changes can be made while running, some of them not, so, the engine itself if who knows the nitty-gritty details.